k-distribution/pl-tutorial/1_k/3_lambda++/lesson_4/tests/free-variable-capture.lambda.out

```diff
@@ -1,11 +1,23 @@
-  V ==K ( a ~> #freezer___LAMBDA0_ ( ( lambda x . lambda y . x ) y z ) )
+  {
+    V:K
+  #Equals
+    a ~> #freezer___LAMBDA_Exp_Exp_Exp0_ ( ( lambda x . lambda y . x ) y z ~> . ) ~> .
+  }
 #Or
-  V ==K ( a ~> #freezer___LAMBDA0_ ( closure ( .Map , x , lambda y . x ) y z ) )
+  {
+    V:K
+  #Equals
+    y ~> #freezer___LAMBDA_Exp_Exp_Exp1_ ( closure ( .Map , x , lambda y . x ) ~> . ) ~> #freezer___LAMBDA_Exp_Exp_Exp0_ ( z ~> . ) ~> #freezer___LAMBDA_Exp_Exp_Exp1_ ( a ~> . ) ~> .
+  }
 #Or
-  V ==K ( y ~> #freezer___LAMBDA1_ ( closure ( .Map , x , lambda y . x ) ) ~> #freezer___LAMBDA0_ ( z ) ~> #freezer___LAMBDA1_ ( a ) )
+  {
+    V:K
+  #Equals
+    y ~> #freezer___LAMBDA_Exp_Exp_Exp1_ ( lambda x . lambda y . x ~> . ) ~> #freezer___LAMBDA_Exp_Exp_Exp0_ ( z ~> . ) ~> #freezer___LAMBDA_Exp_Exp_Exp1_ ( a ~> . ) ~> .
+  }
 #Or
-  V ==K ( y ~> #freezer___LAMBDA1_ ( lambda x . lambda y . x ) ~> #freezer___LAMBDA0_ ( z ) ~> #freezer___LAMBDA1_ ( a ) )
-#Or
-  V ==K ( z ~> #freezer___LAMBDA1_ ( ( lambda x . lambda y . x ) y ) ~> #freezer___LAMBDA1_ ( a ) )
-#Or
-  V ==K ( z ~> #freezer___LAMBDA1_ ( closure ( .Map , x , lambda y . x ) y ) ~> #freezer___LAMBDA1_ ( a ) )
+  {
+    V:K
+  #Equals
+    z ~> #freezer___LAMBDA_Exp_Exp_Exp1_ ( ( lambda x . lambda y . x ) y ~> . ) ~> #freezer___LAMBDA_Exp_Exp_Exp1_ ( a ~> . ) ~> .
+  }
```

Comment:
We have less search results in the post-state than the pre. Is this intentional? (just want to make sure it's double-checked)

Reply:
This is intentional. The java backend implements some custom logic to ensure that heating and cooling rules can apply at any time while also preventing them from applying in an infinitely looping cycle. The Haskell and LLVM backends do not do this, and thus require that any heated term be fully evaluated before cooling. That is why you lose two behaviors in each instance of this particular program. I have previously spoken with @grosu and he has explicitly stated he is okay with making this change for the time being in order to be able to migrate the lessons in the pl-tutorial. We can revisit and readd these behaviors when we have an actual need to implement the full nondeterminism of heating/cooling in K.

Reply:
Ok, is this recorded in an issue somewhere? We don't want to forget.

Reply:
I will create an issue.

Reply:
https://github.com/kframework/llvm-backend/issues/473